src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/endpoints/JurisdictionEndpointTest.java

```diff
@@ -0,0 +1,109 @@
+import feign.Response;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.http.HttpStatus;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.idam.JurisdictionToUserMapping;
+import uk.gov.hmcts.reform.idam.client.IdamClient;
+
+import java.util.Collections;
+import java.util.Map;
+
+import static java.util.AbstractMap.SimpleEntry;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.BDDMockito.never;
+import static org.mockito.BDDMockito.verify;
+import static org.mockito.BDDMockito.willReturn;
+import static org.mockito.BDDMockito.willThrow;
+
+@ExtendWith(MockitoExtension.class)
+public class JurisdictionEndpointTest {
+
+    private static final String VALID_JURISDICTION = "jurisdiction";
+
+    private static final String INVALID_JURISDICTION = "something else";
+
+    private static final Map<String, Map<String, String>> USERS = ImmutableMap.of(
+        VALID_JURISDICTION, ImmutableMap.of(
+            "username", "username",
+            "password","password"
+        ),
+        INVALID_JURISDICTION, ImmutableMap.of(
+            "username", "user",
+            "password", "pass"
+        )
+    );
+
+    @Mock
+    private IdamClient idamClient;
+
+    private JurisdictionEndpoint endpoint;
+
+    @BeforeEach
+    public void setUp() {
+        JurisdictionToUserMapping mapping = new JurisdictionToUserMapping();
+        mapping.setUsers(USERS);
+
+        endpoint = new JurisdictionEndpoint(mapping, idamClient);
+    }
+
+    @DisplayName("Should respond OK for specific jurisdiction which is correctly configured")
+    @Test
+    public void singleValidJurisdiction() {
+        willReturn("token").given(idamClient).authenticateUser("username", "password");
+
+        assertThat(endpoint.jurisdiction(VALID_JURISDICTION)).isEqualTo(HttpStatus.OK);
+    }
+
+    @DisplayName("Should respond INTERNAL_SERVER_ERROR for specific jurisdiction which is incorrectly configured")
+    @Test
+    public void singleInvalidJurisdiction() {
+        willThrow(new RuntimeException("oh no")).given(idamClient).authenticateUser("user", "pass");
+
+        assertThat(endpoint.jurisdiction(INVALID_JURISDICTION)).isEqualTo(HttpStatus.INTERNAL_SERVER_ERROR);
+    }
+
+    @DisplayName("Should respond NOT FOUND for specific jurisdiction which does not exist")
+    @Test
+    public void singleNonExistingJurisdiction() {
+        assertThat(endpoint.jurisdiction("not give access")).isEqualTo(HttpStatus.NOT_FOUND);
+        verify(idamClient, never()).authenticateUser(anyString(), anyString());
+    }
+
+    @DisplayName("Should respond statuses of all configured jurisdictions")
+    @Test
+    public void jurisdictions() {
```

Comment:
I'm ok with DisplayName being more descriptive than method name, as some kind of addition to it, but method name should also be descriptive.